From d48e9090a473fe2a242bdcf28cf3c5a4bcc984f3 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Tue, 17 Jan 2017 08:59:29 +0100
Subject: [PATCH 0940/1345] arm64: dts: cp110: unify gpio aliases and phandles

commit  aa59159f149c1b505edb329d852a2bfe8e4bca5c from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

The gpio phandle from cp110 will be used in upcoming commits for gpio
properties declaration. E.g. interrupt signal of PCA9555 is wired to gpio
62 and it is declared through 'interrupt-parent' and 'interrupts'
properties in pca9555 node via gpio phandle. Instead of using separate
definition in armada-7040.dtsi and armada-8040.dtsi define phandle in
common armada-cp110.dtsi and use it for both armada-7040 and armada-8040
dtsi's

Change-Id: I4b9ef6b13b9bd469990e326e1fae34b3884e93fa
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/38538
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-7040-db.dtsi    |   11 +++++++++--
 arch/arm64/boot/dts/marvell/armada-7040.dtsi       |   12 ++----------
 .../boot/dts/marvell/armada-70x0-customer.dts      |   14 ++++++++++----
 .../boot/dts/marvell/armada-80x0-customer.dts      |   11 +----------
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi      |    4 ++--
 5 files changed, 24 insertions(+), 28 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
index 6abb5b6..2e103a0 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
@@ -76,6 +76,13 @@
 
 	cp110-master {
 		config-space@f2000000 {
+			pinctrl@440000 {
+			compatible = "marvell,a70x0-pinctrl";
+				u3d_vbus_pins: u3d-vbus-pins {
+					marvell,pins = "mpp62";
+					marvell,function = "gpio";
+				};
+			};
 			i2c1: i2c@701000 {
 				pinctrl-names = "default";
 				pinctrl-0 = <&i2c0_pins>;
@@ -100,7 +107,7 @@
 			u3d@520000 {
 				pinctrl-0 = <&u3d_vbus_pins>;
 				pinctrl-names = "default";
-				vbus-gpio = <&gpio2 30 GPIO_ACTIVE_HIGH>;
+				vbus-gpio = <&cpm_gpio1 30 GPIO_ACTIVE_HIGH>;
 			};
 			usb3h0: usb3@500000 {
 				usb-phy = <&usb3h0_phy>;
@@ -195,7 +202,7 @@
 		regulator-min-microvolt = <5000000>;
 		regulator-max-microvolt = <5000000>;
 		enable-active-high;
-		gpio = <&gpio1 12 GPIO_ACTIVE_HIGH>;
+		gpio = <&cpm_gpio0 12 GPIO_ACTIVE_HIGH>;
 		status = "disabled";
 	};
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-7040.dtsi b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
index 37fd2fc..519db1c 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
@@ -53,8 +53,8 @@
 	compatible = "marvell,armada7040", "marvell,armada-ap806-quad",
 		     "marvell,armada-ap806";
 	aliases {
-		gpio1 = &gpio1;
-		gpio2 = &gpio2;
+		gpio1 = &cpm_gpio0;
+		gpio2 = &cpm_gpio1;
 		ethernet0 = &emac0;
 		ethernet1 = &emac2;
 		ethernet2 = &emac3;
@@ -105,18 +105,10 @@
 					marvell,pins = "mpp12";
 					marvell,function = "gpio";
 				};
-				u3d_vbus_pins: u3d-vbus-pins {
-					marvell,pins = "mpp62";
-					marvell,function = "gpio";
-				};
 			};
 			rtc@284000 {
 				status = "okay";
 			};
-			gpio1: gpio@440100 {
-			};
-			gpio2: gpio@440140 {
-			};
 			ppv22@000000 {
 				dma-coherent;
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts b/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts
index 41288fd..c6fde62 100644
--- a/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts
+++ b/arch/arm64/boot/dts/marvell/armada-70x0-customer.dts
@@ -72,8 +72,8 @@
 	#size-cells = <2>;
 
 	aliases {
-		gpio1 = &gpio1;
-		gpio2 = &gpio2;
+		gpio1 = &cpm_gpio0;
+		gpio2 = &cpm_gpio1;
 	};
 
 	/* UPDATE */
@@ -114,6 +114,12 @@
 
 	cpn-110-master {
 		config-space {
+			pinctrl@440000 {
+				u3d_vbus_pins: pca0_pins {
+					marvell,pins = "mpp62";
+					marvell,function = "gpio";
+				};
+			};
 			serial@702000 {
 				status = "disabled";
 			};
@@ -152,7 +158,7 @@
 				/* UPDATE */
 				pinctrl-0 = <&u3d_vbus_pins>;
 				pinctrl-names = "default";
-				vbus-gpio = <&gpio2 30 GPIO_ACTIVE_HIGH>;
+				vbus-gpio = <&cpm_gpio1 30 GPIO_ACTIVE_HIGH>;
 				status = "disabled";
 			};
 			udc@524100 {
@@ -341,7 +347,7 @@
 		regulator-min-microvolt = <5000000>;
 		regulator-max-microvolt = <5000000>;
 		enable-active-high;
-		gpio = <&gpio1 12 GPIO_ACTIVE_HIGH>;
+		gpio = <&cpm_gpio0 12 GPIO_ACTIVE_HIGH>;
 		status = "disabled";
 	};
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts b/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts
index 53f3468..608cb91 100644
--- a/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts
+++ b/arch/arm64/boot/dts/marvell/armada-80x0-customer.dts
@@ -71,11 +71,6 @@
 	#address-cells = <2>;
 	#size-cells = <2>;
 
-	aliases {
-		gpio1 = &gpio1;
-		gpio2 = &gpio2;
-	};
-
 	/* UPDATE */
 	memory@00000000 {
 		device_type = "memory";
@@ -212,10 +207,6 @@
 			comphy {
 				status = "disabled";
 			};
-			gpio1: gpio@440100 {
-			};
-			gpio2: gpio@440140 {
-			};
 			pinctrl@440000 {
 				/* UPDATE */
 				xhci0_vbus_pins: xhci0-vbus-pins {
@@ -239,7 +230,7 @@
 			regulator-min-microvolt = <5000000>;
 			regulator-max-microvolt = <5000000>;
 			enable-active-high;
-			gpio = <&gpio2 15 GPIO_ACTIVE_HIGH>;
+			gpio = <&cpm_gpio1 15 GPIO_ACTIVE_HIGH>;
 			status = "okay";
 		};
 		vccq_sdio_reg: regulator@1 {
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 62a46d0..90cb897 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -64,7 +64,7 @@ pinctrl@440000 {
 	reg = <0x440000 0x20>;
 };
 
-gpio@440100 {
+cpm_gpio0: gpio@440100 {
 	compatible = "marvell,orion-gpio";
 	reg = <0x440100 0x40>;
 	ngpios = <32>;
@@ -79,7 +79,7 @@ gpio@440100 {
 		   <ICU_GRP_NSR 83 IRQ_TYPE_LEVEL_HIGH>;
 };
 
-gpio@440140 {
+cpm_gpio1: gpio@440140 {
 	compatible = "marvell,orion-gpio";
 	reg = <0x440140 0x40>;
 	ngpios = <31>;
-- 
1.7.9.5

